x-pack/plugins/ml/public/application/datavisualizer/index_based/components/field_data_row/document_count_chart/document_count_chart.tsx

```diff
@@ -50,6 +50,21 @@ export const DocumentCountChart: FC<Props> = ({
 
   const dateFormatter = niceTimeFormatter([timeRangeEarliest, timeRangeLatest]);
 
+  const adjustedChartPoints = useMemo(() => {
+    // Display empty chart when no data in range
+    if (chartPoints.length < 1) return [{ time: timeRangeEarliest, value: 0 }];
+
+    // If chart has too few data points
+    // it won't show up unless we pad it on two sides
+    if (chartPoints.length < 3) {
+      return [
+        { time: timeRangeEarliest, value: 0 },
+        ...chartPoints,
+        { time: timeRangeLatest, value: 0 },
+      ];
+    }
+    return chartPoints;
+  }, [chartPoints]);
```

Comment:
it seems like `timeRangeEarliest` is missing  
```suggestion
  }, [chartPoints, timeRangeEarliest]);
```

Reply:
Updated here [`81bf829` (#97655)](https://github.com/elastic/kibana/pull/97655/commits/81bf8292e803086a9276bfdb89f740fdf9c8ba16)